### PR TITLE
Add a simple string set

### PR DESCRIPTION
**Pull request by willsalz**
As titled. Adds a string set that is type'd to a `map[string]bool` and has basic `Init`, `Contains` and `Add` methods as well as a function `Intersection` that returns a new Set that is the intersection of two input `Set`s.

### CODE REVIEW

**Comment by willsalz** [comment]
cc @anthonybishopric @bkoltai @macrael

**Comment by bkoltai** [comment]
GitHub now has multi-assignment FYI

**Comment by bkoltai** [comment]
https://github.com/deckarep/golang-set ?

**Comment by willsalz** [comment]
No deps!

File: pandemic/set.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package pandemic
+
+type Set map[string]bool
```
Comment (anthonybishopric) [inline pandemic/set.go:3]: I think there's a tiny performance improvement by using an empty struct (`struct{}`) than using `bool`.

**Comment by anthonybishopric** [comment]
👍

**Pull request merged**